interface/src/app/gene/genes.js

```diff
@@ -0,0 +1,384 @@
+      });
+    },
+    replace: true,
+    restrict: "E",
+    scope: {
+      query: '@'
+    },
+    templateUrl: 'gene/no-result-button.tpl.html'
+  };
+}])
+
+// Directive for button with a gene, should add gene
+// and remove entire row from list
+.directive('geneResultButton', ['SearchResults', 'SelectedGenesFactory',
+  function(SearchResults, SelectedGenesFactory) {
+    return {
+      restrict: "E",
+      link: function(scope, element, attr) {
+        element.bind("click", function() {
+          scope.$apply(function() {
+            SelectedGenesFactory.addGene(scope.gene);
+            SearchResults.remove(scope.query);
+          });
+        });
+      },
+      replace: true,
+      templateUrl: 'gene/gene-result-button.tpl.html'
+    };
+  }
+])
+
+// Directive for button with a gene, should add gene
+// and remove entire row from list
+.directive('selectedGeneButton', ['SelectedGenesFactory',
+  function(SelectedGenesFactory) {
+    return {
+      restrict: "E",
+      link: function(scope, element, attr) {
+        element.bind("click", function() {
+          scope.$apply(function() {
+            SelectedGenesFactory.removeGene(scope.gene);
+          });
+        });
+      },
+      replace: true,
+      templateUrl: 'gene/gene-result-button.tpl.html'
+    };
+  }
+])
+
+// Directive for button to get more options, should get
+// next page of search results for this query from the server
+.directive('moreResultButton', ['SearchResults', function(SearchResults) {
+  return {
+    link: function(scope, element, attr) {
+      element.bind("click", function() {
+        scope.page = scope.pageDict.page + 1;
+        scope.$apply(function() {
+          scope.pageDict.page = scope.page;
+          scope.updatePage(scope.page);
+        });
+      });
+    },
+    replace: true,
+    restrict: "E",
+    scope: false,
+    templateUrl: 'gene/more-result-button.tpl.html'
+  };
+}])
+
+// Directive for button to get previous search results 
+.directive('previousResultButton', ['SearchResults', function(SearchResults) {
+  return {
+    link: function(scope, element, attr) {
+      element.bind("click", function() {
+        scope.page = scope.pageDict.page - 1;
+        scope.$apply(function() {
+          scope.pageDict.page = scope.page;
+          scope.updatePage(scope.page);
+        });
+      });
+    },
+    replace: true,
+    restrict: "E",
+    scope: false,
+    templateUrl: 'gene/previous-result-button.tpl.html'
+  };
+}])
+
+// Directive for search buttonset, has buttons for handling
+// search results
+.directive('searchButtonset', ['SearchResults', function(SearchResults) {
+  return {
+    controller: function($scope) {
+      $scope.pageDict = {
+        page: 1
+      };
+      $scope.results = SearchResults.getQueryResults($scope.query);
+      $scope.found = $scope.results.found;
+
+      var begin, end;
+      $scope.updatePage = function(page) {
+        begin = (page - 1) * 3;
+        end = begin + 3;
```

Comment:
This is the number of gene results per 'page' in the button-set with the gene results. I will make a variable for this so that we can change it in the future if we want it to be something other than 3.